box/boxunreal.py

```diff
@@ -87,6 +99,9 @@ def take_action(self) -> tuple[Action, Action]:
             RuntimeError: If the action is not defined.
         """
 
+        # We set the raycast length here to ensure the checked movement forward is being correctly compared.
+        self.ue5.set_raycast_length(self.raycast_length)
```

Comment:
I don’t think this needs to be called every time we call `take_action`

Reply:
I moved it to where we initialize our class.

Reply:
However, it turns out this doesn't quite work. It just keeps the same default value if placed there.

Reply:
It worked once I placed it outside of boxunreal.py and into boxsim.py right before we call take_action